vscode/microsoft-kiota/src/utilities/deep-linking.ts

```diff
@@ -103,14 +109,38 @@ export function validateDeepLinkQueryParams(queryParameters: Partial<Integration
     errormsg.push("Invalid parameter 'type' deeplinked. Expected values: " + acceptedPluginTypes.join(","));
   }
 
+  if (projectPath && !path.isAbsolute(projectPath)) {
+    projectPath = undefined;
+    errormsg.push(`A relative paths is not supported for the projectPath parameter`);
+  }
+
   validQueryParams = {
-    descriptionurl: descriptionurl,
-    name: name,
+    descriptionurl,
+    name,
     kind: validKind,
     type: providedType,
     language: givenLanguage,
-    source: source,
-    ttkContext: queryParameters.ttkContext ? queryParameters.ttkContext : undefined
+    source,
+    ttkContext: queryParameters.ttkContext ? queryParameters.ttkContext : undefined,
+    projectPath
   };
   return [validQueryParams, errormsg];
 }
+
+function determineOutputPath(deepLinkParams: Partial<IntegrationParams>): string | undefined {
+  if (deepLinkParams.projectPath) {
+    try {
+      if (!fs.existsSync(deepLinkParams.projectPath)) {
```

Comment:
Use async APIs for file system access. Using sync APIs might block the event loop and make the extension unresponsive